### ISSUE TITLE
Inconsistent parameter naming Macro __DEFINE_HAS_LVAL_MMBR_FUNC_ 7

### ISSUE THREAD

**Issue by AnantaYudica** [AnantaYudica/basic#15]
```
#include "macro/HasLValueMemberFunction.h"

struct B
{};

template<typename>
struct C
{
    void Foo1() {}
};

template<>
struct C<std::false_type>
{};

struct D
{
    template<typename T>
    C<T> Foo3() {return C<T>();}
};

/**
 *  template<typename T, typename T1, template<typename> class Tr,
 *      typename... Targs>
 *  constexpr auto _HasLValMmbrFunc7(T t, Targs... args) ->
 *      decltype(Tr<T1>(t.template Foo3<T1>(args...)), std::true_type());
 *  template<typename T, typename T1, template<typename> class Tr,
 *      typename... Targs>
 *  constexpr std::false_type _HasLValMmbrFunc7(...);
 *  template<typename T, typename T1, typename Tr, typename... Targs>
 *  struct HasLValMmbrFunc7
 *  {
 *      typedef bool value_type;
 *      typedef bool ValueType;
 *      static constexpr value_type value = decltype(_HasLValMmbrFunc7<
 *          typename basic::type::trait::rem::ptr::Recursive<T>::type,
 *                  T1, Tr, Targs...>(
 *              std::declval<typename std::remove_cv<typename basic::type::
 *                  trait::rem::ptr::Recursive<T>::type>::type>(), 
 *              std::declval<typename std::remove_cv<Targs...>::type>()
 *          ))::value;
 *      static constexpr ValueType Value = decltype(_HasLValMmbrFunc7<
 *          typename basic::type::trait::rem::ptr::Recursive<T>::type,
 *                  T1, Tr, Targs...>(
 *              std::declval<typename std::remove_cv<typename basic::type::
 *                  trait::rem::ptr::Recursive<T>::type>::type>(), 
 *              std::declval<typename std::remove_cv<Targs...>::type>()
 *          ))::value;
 *  };
 * */

#define __DEFINE_HAS_LVAL_MMBR_FUNC_PROTO_FUNC_TMPL_ARG_LIST_PFTAL1(\
    __T_NAME__, \
    __T_RET_NAME__,\
    __T_ARGS_NAME__,\
    __T_NAME_REM_RECV_PTR_TYPE__, ...)\
__T_NAME_REM_RECV_PTR_TYPE__,##__VA_ARGS__, T1, __T_RET_NAME__, __T_ARGS_NAME__...

#define __DEFINE_HAS_LVAL_MMBR_FUNC_TMPL_PARAM_LIST_TPL3(\
    __T_NAME__,\
    __T_RET_NAME__,\
    __T_ARGS_NAME__)\
typename __T_NAME__, typename T1, template<typename> class __T_RET_NAME__,\
    typename... __T_ARGS_NAME__

#define __DEFINE_HAS_LVAL_MMBR_FUNC_PROTO_FUNC_TRUE_TMPL_PARAM_LIST_PFTTPL3(\
    __T_NAME__,\
    __T_RET_NAME__,\
    __T_ARGS_NAME__)\
typename __T_NAME__, typename T1, template<typename> class __T_RET_NAME__,\
    typename... __T_ARGS_NAME__

#define __DEFINE_HAS_LVAL_MMBR_FUNC_PROTO_FUNC_FALSE_TMPL_PARAM_LIST_PFFTPL3(\
    __T_NAME__,\
    __T_RET_NAME__,\
    __T_ARGS_NAME__)\
typename __T_NAME__, typename T1, template<typename> class __T_RET_NAME__,\
    typename... __T_ARGS_NAME__

#define __DEFINE_HAS_MMBR_FUNC_PROTO_FUNC_T_RET_NAME_TMPL_ARG_LIST_PFTRNRAL1()\
T1

__DEFINE_HAS_LVAL_MMBR_FUNC_(HasLValMmbrFunc7, _HasLValMmbrFunc7, T, Tr, 
    Targs, TPL3, T_NAME_,, PFTTPL3, PFFTPL3, PFTAL1, PROTO_FUNC_T_NAME_,,\
    PROTO_FUNC_T_RET_NAME_TMPL_, PFTRNRAL1, template Foo3<T1>);

```
parameter macro using name `__PROTO_FUNC_T_RET_NAME_TMPL_ARG_LIST_POSTFIX__` in `__DEFINE_HAS_LVAL_MMBR_FUNC_`
but call back implementation using name `__DEFINE_HAS_MMBR_FUNC_PROTO_FUNC_T_RET_NAME_TMPL_ARG_LIST_`

Similar with #4, #9, #10, #11, #12, #13 and #14

**Comment by AnantaYudica** [comment]
Fixed in v0.1-beta-2

**Comment by AnantaYudica** [comment]
Not a bug, Macro `__DEFINE_HAS_RVAL_MMBR_FUNC_` will be deleted in v0.1-beta-2.
